### PR TITLE
Rename ObjectRef methods to be consistent and predictable

### PR DESCRIPTION
**Pull request by rubenwardy**

### CODE REVIEW

File: src/script/lua_api/l_internal.h [hunk 1]
```diff
@@ -30,6 +30,7 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include "common/c_internal.h"
 
 #define luamethod(class, name) {#name, class::l_##name}
+#define luamethod_bc(class, name, alias) {#name, class::l_##name}, {#alias, class::l_##name}
```
Comment (sfan5) [inline src/script/lua_api/l_internal.h]: `bc` ?

Comment (rubenwardy) [inline src/script/lua_api/l_internal.h]: backwards compatible.

I'm unsure about this name. Feel free to suggest another one, ideally 9 characters long to match `luamethod`

Comment (sfan5) [inline src/script/lua_api/l_internal.h]: `luamethod_aliased` would be much clearer (or even just `luamethod_alias`)

Comment (rubenwardy) [inline src/script/lua_api/l_internal.h]: I'd prefer aliased over alias, as alias implies it's just an alias function whereas alias implies it's a luamethod which is also aliased

**Comment by est31** [comment]
I think we should also add deprecation warnings (via the setting dependent deprecation warning system that we already have) so that people move over to the new naming scheme.

**Comment by tenplus1** [comment]
Personally I find the deprecation warnings annoying and players who use mods often think something is wrong...  Add a forum thread for engine and function changes to warn modders instead.

**Pull request merged**